event/src/anchor.rs

```diff
@@ -0,0 +1,86 @@
+use cid::Cid;
+use indexmap::IndexMap;
+use serde::{Deserialize, Serialize};
+
+use crate::unvalidated::{Proof, RawTimeEvent};
+
+/// AnchorRequest for a Data Event on a Stream
+pub struct AnchorRequest {
+    /// The CID of the stream
+    pub id: Cid,
+    /// The CID of the Event to be anchored
+    pub prev: Cid,
+}
+
+/// Merkle tree node
+pub type MerkleNode = Vec<Cid>;
```

Comment:
what about a node that has only has a child in one direction?  Don't we really need each node to be a pair of Option<Cid> so we can also use None when one branch is empty?

Reply:
for parsing yes for building the tree we don't have any not full nodes.

Reply:
how is that possible?  What if you have a number of events that isn't a power of 2?  Won't you always have some unbalanced nodes?

Reply:
![image](https://github.com/user-attachments/assets/d894c47f-6175-4493-bffe-8a130f8b3f05)
The depth may not be balanced but there are no nulls.
The peeks are the place values so
peek[0] has 1 anchor requested CID or a null.
peek[1] has 2 anchor requested CID or a null.
peek[2] has 4 anchor requested CID or a null.
peek[3] has 8 anchor requested CID or a null.
peek[n] has 2**n anchor requested CID or a null.

then we just role up the side merging the pow2 trees.
If we kept the nulls we would get balanced depth but nulls in the roll up nodes.
Since we don't bother with the nodes that have an empty branch we get no nulls but shorter paths where the half empty nodes would have been,

Reply:
Ah this is still with the MMR approach, I was assuming we switched to the simpler balanced tree building approach